core/tests/create_snapshot.rs

```diff
@@ -56,7 +54,43 @@ async fn create_snapshot_works() {
         .expect("Failed to create a tempdir");
     let snap_file_path = temp_dir.path().join("snapshot.snap");
 
-    common::run_with_timeout(Duration::from_secs(60), async move {
+    common::run_with_timeout(Duration::from_secs(360), async move {
+        async fn block_hash(block_number: u64, url: &str, timeout: u64) -> Result<Hash, String> {
+            use substrate_rpc_client::{ws_client, ChainApi};
+            use sp_rpc::{list::ListOrValue, number::NumberOrHex};
+            use node_primitives::{Header};
+
+            let start = std::time::Instant::now();
+            let mut interval = 27;
+
+            let rpc = loop {
+                match ws_client(url).await {
+                    Ok(rpc) => break rpc,
+                    Err(err) => {
+                        let elapse = start.elapsed();
+                        if elapse > Duration::from_secs(timeout) {
+                            panic!("Failed to connect within {:?}: {}", elapse, err);
+                        }
+                    }
+                }
+                tokio::time::sleep(Duration::from_secs(interval)).await;
+                interval = (interval / 3 * 2).max(8);
+            };
+
+            let result = ChainApi::<(), Hash, Header, ()>::block_hash(
+                &rpc,
+                Some(ListOrValue::Value(NumberOrHex::Number(block_number))),
+            )
+            .await
+            .map_err(|_| "Couldn't get block hash".to_string())?;
```

Comment:
What I mean is by the time you get there, the block may not have been mined yet so probably needs to continue waiting *after* it has connected until confirmation that the target block is mined